com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -259,6 +259,10 @@ public void LazyInitialize()
             ResetData();
             Initialize();
             InitializeSensors();
+            if (Academy.Instance.FirstAcademyReset)
```

Comment:
There's no way I won't say "WTF" if I look at this code a month from now. Can we make it clearer in comments (either here or in OnEpisodeBegin) that OnEpisodeBegin will be always be called exactly once on the Agent before it starts observing/acting, either during Academy.ForcedFullReset() _or_ Agent.LazyInitialize().